### DISCUSSION TITLE
What happens if memcached in unavailable?

### DISCUSSION THREAD

**Discussion by ngdaddikar** [express-rate-limit/rate-limit-memcached#8] [Q&A] [answered]
Hi, 

If memcached server is down how does the express rate limiter behave? Does it give an error? Is there a "fallback" mechanism?

Thanks.

**Comment by nfriedly** [comment]
It will go to Express's error handler - http://expressjs.com/en/guide/error-handling.html#the-default-error-handler

There was some discussion of making a configuration option to "fail open" and allow unlimited requests through if the datastore goes down. This would be in express-rate-limit so that it could apply to all data stores, but no work has been done on that yet.

**Comment by ngdaddikar** [comment]
I am new to node-js. Is there any example on how to bypass errors in the the rate limiter i.e let all requests go through when the store is down. Thanks.

**Reply by nfriedly** [reply, marked as answer]
You could write your own wrapper, something like this:

```js
const limiter = rateLimit({ /* ... */ });
app.use((req, res, next) => {
  limiter(req, res, (err) => {
    if (err) {
      console.error('error from express-rate-limit, allowing request', err);
    }
    next();
  }
});
```

That will log the error and then continue handling the request as if the limiter was allowed. It's not a *great* idea, because if a malicious user figured out a way to reliably trigger errors, they'd be able to bypass your limiter. (Also, this example ignores all errors, not just store-related ones.) But, it would allow you to prioritize availability over rate-limiting.
